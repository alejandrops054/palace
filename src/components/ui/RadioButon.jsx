import React from 'react';
import '../../css/ui/RadioButon.css';
//label:etiqueta de tipo String

const RadioButon = ({label, col }) =>{
    return(
        <div className={`col-lg-${col} mb-3`}>
            <label>{label}</label>
            <div className="onoffswitch" >
                <input type="checkbox" name="onoffswitch" className="onoffswitch-checkbox" id="myonoffswitch" tabindex="0" checked ></input>
                <label className="onoffswitch-label" for="myonoffswitch">
                    <span className="onoffswitch-inner"></span>
                    <span className="onoffswitch-switch"></span>
                </label>
            </div>
        </div>
        
    )
}

export default RadioButon