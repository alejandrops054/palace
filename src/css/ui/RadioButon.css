/* Ancho del Switch */
.onoffswitch {
    position: relative; width:80px;
    -webkit-user-select:none; -moz-user-select:none; -ms-user-select: none;
}

/* Quita el Estilo del Switch Original del HTML */
.onoffswitch-checkbox {
    position: absolute;
    opacity: 0;
    pointer-events: none;
}

/* Grueso del Border */
.onoffswitch-label {
    display: block; overflow: hidden; cursor: pointer;
    border: 3px solid #999999; border-radius: 20px;
}

/* Estilo del Switch Apagado */
.onoffswitch-inner {
    display: block; width: 200%; margin-left: -100%;
    transition: margin 0.3s ease-in 0s;
}
/* Alto del Switch */
.onoffswitch-inner:before, .onoffswitch-inner:after {
    display: block; float: left; width: 50%; height: 30px; padding: 0; line-height: 30px;
    font-size: 14px; color: white; font-family: Trebuchet, Arial, sans-serif; font-weight: bold;
    box-sizing: border-box;
}

/* Estilo de la Etiqueta ON */
.onoffswitch-inner:before {
    content: "ON";
    padding-left: 10px;
    background-color: #4234c1; color: #FFFFFF;
}
/* Estilo de la Etiqueta OFF */
.onoffswitch-inner:after {
    content: "OFF";
    padding-right: 10px;
    background-color: #EEEEEE; color: #999999;
    text-align: right;
}

/* Estilo del Punto */
.onoffswitch-switch {
    display: block; width: 30px; margin: 6px;
    background: #FFFFFF;
    position: absolute; top: 0; bottom: 0;
    right: 56px;
    border: 2px solid #999999; border-radius:20px;
    transition: all 0.3s ease-in 0s; 
}

.onoffswitch-checkbox:checked + .onoffswitch-label .onoffswitch-inner {
    margin-left: 0px;
}

.onoffswitch-checkbox:checked + .onoffswitch-label .onoffswitch-switch {
    right: 0px; 
}