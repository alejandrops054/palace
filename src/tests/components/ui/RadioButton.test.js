import React from 'react';
import { shallow } from 'enzyme'
import RadioButon from '../../../components/ui/RadioButon'

describe('Prueba componente UI <RadioButton/>', () => {
    test('Componente recibe props label:String, col:Number', () => {
        const store = { 
            titulo:'Titulo de etiqueta',
            col:6
        }
        const wrapper = shallow(
            <RadioButon
                label={store.titulo}
                col={store.col}
            />
        );

        expect(wrapper).toMatchSnapshot();
    })

})
