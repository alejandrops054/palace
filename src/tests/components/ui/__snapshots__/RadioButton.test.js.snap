// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba componente UI <RadioButton/> Componente recibe props label:String, col:Number 1`] = `
<div
  className="col-lg-6 mb-3"
>
  <label>
    Titulo de etiqueta
  </label>
  <div
    className="onoffswitch"
  >
    <input
      checked={true}
      className="onoffswitch-checkbox"
      id="myonoffswitch"
      name="onoffswitch"
      tabindex="0"
      type="checkbox"
    />
    <label
      className="onoffswitch-label"
      for="myonoffswitch"
    >
      <span
        className="onoffswitch-inner"
      />
      <span
        className="onoffswitch-switch"
      />
    </label>
  </div>
</div>
`;
